algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/treatment/HasHadSystemicTreatmentInAdvancedOrMetastaticSetting.kt

```diff
@@ -12,86 +13,92 @@ import com.hartwig.actin.datamodel.clinical.treatment.history.Intent
 import com.hartwig.actin.datamodel.clinical.treatment.history.TreatmentHistoryEntry
 import java.time.LocalDate
 
-class HasHadSystemicTreatmentInAdvancedOrMetastaticSetting(private val referenceDate: LocalDate) : EvaluationFunction {
+class HasHadSystemicTreatmentInAdvancedOrMetastaticSetting(private val referenceDate: LocalDate, private val metastaticOnly: Boolean) :
+    EvaluationFunction {
     override fun evaluate(record: PatientRecord): Evaluation {
         val priorTreatments = record.oncologicalHistory.sortedWith(TreatmentHistoryAscendingDateComparator())
         val priorSystemicTreatments = priorTreatments.filter { it.treatments.any(Treatment::isSystemic) }
-        val (curativeTreatments, nonCurativeTreatments) = priorSystemicTreatments.partition { it.intents?.contains(Intent.CURATIVE) == true }
-        val (recentNonCurativeTreatments, nonRecentNonCurativeTreatments) = partitionRecentTreatments(nonCurativeTreatments, false)
-        val (recentNonCurativeTreatmentsIncludingUnknown, _) = partitionRecentTreatments(nonCurativeTreatments, true)
-        val nonCurativeTreatmentsWithUnknownStopDate = nonCurativeTreatments.filter { it.stopYear() == null }
+        val wrongIntents = if (metastaticOnly) Intent.curativeAdjuvantNeoadjuvantSet() else setOf(Intent.CURATIVE)
+        val (wrongIntentTreatments, potentiallyCorrectIntentTreatments) =
```

Comment:
Terms "wrong" and "correct" don't feel quite right here.  Perhaps "excluded" and "included"?